pkg/securitycontext/admission/admission.go

```diff
@@ -0,0 +1,284 @@
+			if constraint.SELinuxContext.SELinuxOptions == nil {
+				constraint.SELinuxContext.SELinuxOptions = &kapi.SELinuxOptions{}
+			}
+
+			level, err := getPreallocatedLevel(namespace)
+			if err != nil {
+				return err
+			}
+			constraint.SELinuxContext.SELinuxOptions.Level = level
+		}
+
+		provider, err := scc.NewSimpleProvider(constraint, c.client)
+		if err != nil {
+			return err
+		}
+		providers[constraint.Name] = provider
+	}
+
+Containers:
+	for i, container := range pod.Spec.Containers {
+		for providerName, provider := range providers {
+			// Create a security context for this provider
+			context, err := provider.CreateSecurityContext(pod, &container)
+			if err != nil {
+				return err
+			}
+
+			// Validate the copy of the container against the security context provider
+			container.SecurityContext = context
+			errs := provider.ValidateSecurityContext(pod, &container)
+			if len(errs) != 0 {
+				continue
+			}
+
+			// Set the security context on actual container
+			pod.Spec.Containers[i].SecurityContext = context
+			if pod.Annotations == nil {
+				pod.Annotations = make(map[string]string, 0)
+			}
+			pod.Annotations[validatedSCCAnnotation] = providerName
+			continue Containers
+		}
+
+		// If we have reached this code, we couldn't find an SCC that matched
+		// the requested security context for this container
+		//
+		// TODO: better error message
+		return errors.NewBadRequest("Unable to find a valid security context constraint for container")
+	}
+
+	return nil
+}
+
+// getMatchingSecurityContextConstraints returns constraints from the store that match the group,
+// uid, or user of the service account.
+func (c *constraint) getMatchingSecurityContextConstraints(ns string, serviceAccount *kapi.ServiceAccount) ([]*kapi.SecurityContextConstraints, error) {
+	userInfo := serviceaccount.UserInfo(ns, serviceAccount.Name, string(serviceAccount.UID))
+	matchedConstraints := make([]*kapi.SecurityContextConstraints, 0)
+
+	for _, c := range c.store.List() {
+		constraint, ok := c.(*kapi.SecurityContextConstraints)
+		if !ok {
+			return nil, errors.NewInternalError(fmt.Errorf("Error converting object from store to a security context constraint: %v", c))
+		}
+		for _, userGroup := range userInfo.GetGroups() {
+			if constraintSupportsGroup(userGroup, constraint.Groups) {
+				matchedConstraints = append(matchedConstraints, constraint)
+				break
+			}
+		}
+
+		for _, user := range constraint.Users {
+			if userInfo.GetName() == user || userInfo.GetUID() == user {
+				matchedConstraints = append(matchedConstraints, constraint)
+				break
+			}
+		}
+	}
+
+	return matchedConstraints, nil
+}
+
+// getPreallocatedUIDRange retrieves the annotated value from the service account, splits it to make
+// the min/max and formats the data into the necessary types for the strategy options.
+func getPreallocatedUIDRange(ns *kapi.Namespace) (min *int64, max *int64, err error) {
```

Comment:
don't use named return args... makes it hard to follow what is actually being returned